openapi2beans/pkg/generator/schema2package.go

```diff
@@ -0,0 +1,200 @@
+
+			constDataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+
+			constantDataMembers = append(constantDataMembers, &constDataMember)
+
+		} else {
+
+			dataMember := DataMember {
+				Name: name,
+				CamelCaseName: convertToCamelCase(name),
+				MemberType: propertyToJavaType(property),
+				Description: description,
+				ConstantVal: constVal,
+			}
+			dataMembers = append(dataMembers, &dataMember)
+				
+			if property.IsSetInConstructor() {
+				requiredMember := RequiredMember {
+					DataMember: &dataMember,
+				}
+				requiredMembers = append(requiredMembers, &requiredMember)
+			}
+		}
+		
+	}
+	sort.SliceStable(dataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(dataMembers[i], dataMembers[j])})
+	sort.SliceStable(requiredMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(requiredMembers[i].DataMember, requiredMembers[j].DataMember)})
+	sort.SliceStable(constantDataMembers, func (i int, j int) bool {return isDataMemberLessThanComparison(constantDataMembers[i], constantDataMembers[j])})
+	if requiredMembers != nil {
+		requiredMembers[0].IsFirst = true
+	}
+	return dataMembers, requiredMembers, constantDataMembers
+}
+
+func propertyToJavaType(property *Property) string {
+	javaType := ""
+	if property.IsReferencing() || property.typeName == "object" || property.IsEnum() {
+		javaType = property.resolvedType.name
+	} else {
+		if property.typeName == "string" {
+			javaType = "String"
+		} else if property.typeName == "integer" {
+			javaType = "int"
+		} else if property.typeName == "number" {
+			javaType = "double"
+		}else if property.typeName == "" {
+			javaType = "Object"
+		} else {
+			javaType = property.typeName
+		}
+	}
+
+	if property.IsCollection() {
+		dimensions := property.cardinality.max / 128
+		for range dimensions {
+			javaType += "[]"
+		}
+	}
+
+	return javaType
+}
+
+func convertToCamelCase(name string) string {
+	initialLetter := name[0]
+	camelCaseName := strings.ToUpper(string(initialLetter)) + name[1:]
+	return camelCaseName
+}
+
+func convertToConstName(name string) string {
+	var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+	var matchAllCap   = regexp.MustCompile("([a-z0-9])([A-Z])")
+
+	constName := matchFirstCap.ReplaceAllString(name, "${1}_${2}")
+    constName  = matchAllCap.ReplaceAllString(constName, "${1}_${2}")
+
+    return strings.ToUpper(constName)
+}
+
+func convertConstValueToJavaReadable(constVal string, constType string) string {
+	if constType == "string" {
+		constVal = "\"" + constVal + "\""
+	}
+	return constVal
+}
+
+// function used for sorting, groups variables by type and then alphabetically
+// order of variables is:
+// boolean > int > double > String > other
+func isDataMemberLessThanComparison(dataMember *DataMember, comparisonMember *DataMember) bool {
```

Comment:
Wondering now if we should be sorting, rather than preserving the order they appear in the yaml. Hmm. 
Sorting by type then alphabetically seems a bit 'odd' but sorting actually seems a bit 'odd' also the more I think about it.
Given that the bean is rendered to json, I don't think it matters functionally, just when people are comparing the bean vs the yaml...
As I'm not sure, perhaps we should ask more widely, like ask savvas/eamon/jade ?

Reply:
I've found that due to the yaml being converted into a map it completely looses the original structure. this is ordered specifically for test cases so they don't randomly fail, and the order is entirely arbitrary and the way I personally prefer laying out my code, I am absolutely open to changing it though.

Reply:
I think the trouble may be that if someone defines a structure in the yaml, and two properties are related, then having the order changed may make it more confusing rather than less.

Are you saying the reading of the yaml turns it into a map ? In which case you have no control over it, otherwise I would suggest an OrderedMap, and add an ordinal which can be used as a sort code.

If you have no control over it, then ordering is arbitrary I agree, and resolve this conversation please. Glad it's ordered in some way anyway :-)